src/bindings/c/include/openvino/c/ov_tensor.h

```diff
@@ -24,6 +24,38 @@ typedef struct ov_tensor ov_tensor_t;
  * @{
  */
 
+/**
+ * @struct ov_tensor_list_t
+ * @brief A list of tensors
+ */
+typedef struct ov_tensor_list ov_tensor_list_t;
+
+/**
+ * @brief Creat a tensor list
+ * @ingroup tensor
+ * @param tensors A point to ov_tensor_list_t
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_tensor_list_create(ov_tensor_list_t** tensors);
+
+/**
+ * @brief Add a tensor to a list
+ * @ingroup tensor
+ * @param tensor A point to the tensor will be added to tensor list
+ * @param tensors A point to tensor list
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_tensor_list_add(ov_tensor_list_t* tensors, const ov_tensor_t* tensor);
```

Comment:
I don't believe honestly that somebody will add tensors to the list. I think we can provide more C aligned API and allow user to use C arrays.

Reply:
If the tensor list size is a fixed value, it is fine to use such C array. But if size of tensor list is dynamic, C array cannot support.
@ilyachur How do you think?

Reply:
@riverlijunjie  do you have an example where model have a dynamic set of inputs or outputs?
Different models can have different numbers of inputs, but the size of inputs and outputs at the current moment is always constant and cannot be changed in the runtime after the compilation of the model.
I think it is not a big deal to allocate the memory for all input/output tensors from the model before the filling tensors.

Reply:
@ilyachur I got your point and agree with you. Considered that the tensor list size always is fixed, we don't need use a dynamic size tensor list definition.
@zhaixuejun1993  let's follow Ilya's idea :)